File: Jump game/sol.c
bool canJump(int* nums, int numsSize) {
    int val = 0;
    int max(int a, int b)
    {
        if(a>b)
        {
            return a;
        }
        else
        {
            return b;
        }
    }
    for(int i=0;i<numsSize;i++)
    {
        if(i > val)
        {
            return false;
        }
        val = max(val, i+nums[i]);
    }
    return true;
}

File: Jump game 2/sol.py
class Solution(object):
    def jump(self, nums):
        size = len(nums)
        min_jump, current, farthest = 0, 0, 0
        for i in range(0, size-1):
            farthest = max(farthest, i + nums[i])
            if(i == current):
                min_jump += 1
                current = farthest
        return min_jump

        
